vllm/model_executor/layers/quantization/input_quant_fp8.py

```diff
@@ -70,27 +82,19 @@ def forward_native(
         scale: Optional[torch.Tensor] = None,
         scale_ub: Optional[torch.Tensor] = None,
     ):
+        if self.is_group_quant:
+            assert scale is None, "Group quantization is always dynamic"
+            return self._quantize_group_native(x)
+
         assert (scale is not None) == self.static
         assert scale_ub is None or (not self.static and self.group_shape
                                     == GroupShape.PER_TOKEN
                                     and scale_ub.numel() == 1)
 
-        if scale is None:
-            if self.group_shape == GroupShape.PER_TOKEN:
-                x_max, _ = x.abs().max(dim=-1)
-                x_max = x_max.unsqueeze(-1).to(torch.float32)
-                if scale_ub is not None:
-                    x_max = x_max.clamp(max=scale_ub)
-            else:
-                x_max = x.abs().max().unsqueeze(-1).to(torch.float32)
-
-            scale = x_max / _FP8_MAX
-            scale = scale.clamp(min=_FP8_MIN_SCALING_FACTOR)
-
-        # Even for dynamic per-token scales,
-        # reciprocal performs slightly better than division
-        out = x.to(torch.float32) * scale.reciprocal()
-        out = out.clamp(_FP8_MIN, _FP8_MAX).to(_FP8_DTYPE)
+        if self.use_per_token_if_dynamic and scale is None:
+            out, scale = quantize_fp8_per_token(x, scale, scale_ub)
+        else:
+            out, scale = quantize_fp8_per_tensor(x, scale)
```

Comment:
I think this adds an unnecessary layer of indirection. We don't need to extract the torch implementations out of this file. If you want to add utility methods to this class that's fine but no need to move this elsewhere